pkg/requestgen/generator.go

```diff
@@ -140,3 +179,27 @@ func keyUsagesFromAttributes(usagesCSV string) []cmapi.KeyUsage {
 
 	return keyUsages
 }
+
+// executeTemplate executes the template on the given csv with volume context
+// provided by the metadata.
+func executeTemplate(meta metadata.Metadata, csv string) (string, error) {
+	ptmpl, err := template.New("").Parse(csv)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse dnsNames for templating: %w", err)
+	}
+
+	var buf bytes.Buffer
+	if err := ptmpl.Execute(&buf, struct {
+		PodName      string
+		PodNamespace string
+		PodUID       string
+	}{
+		PodName:      meta.VolumeContext["csi.storage.k8s.io/pod.name"],
+		PodNamespace: meta.VolumeContext["csi.storage.k8s.io/pod.namespace"],
+		PodUID:       meta.VolumeContext["csi.storage.k8s.io/pod.uid"],
```

Comment:
These could perhaps be made constants, not a blocker though

Reply:
Yeah; these keys aren't exposed as a const in k8s.io unfortunately. Will add these as local consts.